src/Octoshift/Services/GithubApi.cs

```
@@ -1089,24 +1090,102 @@ public virtual async Task<string> UploadArchiveToGithubStorage(string orgDatabas
+        {
+            nextUrl = await UploadPartAsync(buffer, bytesRead, nextUrl.ToString());
+        }
+
+        // 3. Complete the upload
+        await CompleteUploadAsync(nextUrl.ToString());
+
+        return $"gei://archive/{guid}";
+    }
+
+    private async Task<(string Response, KeyValuePair<string, IEnumerable<string>>[] Headers)> StartUploadAsync(string uploadUrl, string archiveName, long contentSize)
+    {
+        // Create the body for the request
+        var body = new JObject
+        {
+            ["content_type"] = "application/octet-stream",
+            ["name"] = archiveName,
+            ["size"] = contentSize
+        };
+
+        // Post the request and get both response content and headers as a tuple
+        var (response, headers) = await _client.PostTupleAsync(uploadUrl, body);
+
+        // Return the tuple with response and headers
+        return (response, headers);
+    }
+
+
+    private async Task<Uri> UploadPartAsync(byte[] body, int bytesRead, string nextUrl)
+    {
+        var content = new ByteArrayContent(body, 0, bytesRead);
+        content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
+        var (response, headers) = await _client.PatchTupleAsync(nextUrl, content);
+
+        return GetNextUrl(headers);
+    }
+
+    private async Task CompleteUploadAsync(string lastUrl)
+    {
+        // Create an empty HttpContent object
+        var content = new StringContent(string.Empty);
+
+        // Set the correct Content-Type header (adjust based on API requirements)
+        content.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
+
+        await _client.PutAsync(lastUrl, content);
+    }
+
+    private Uri GetNextUrl(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers)
+    {
+        // Find the Location header
+        var locationHeader = headers.FirstOrDefault(header => header.Key == "Location");
```

Do we expect the header to be always present here? If yes then we should probably raise an exception or just use `First` so it will raise an exception if there header is not there.